src/KahanSummation.jl

```diff
@@ -115,6 +115,7 @@ function sum_kbn(A)
 end
 
 function sum_kbn(f::Function, A)
+    f === identity && return sum_kbn(A)
```

Comment:
This is backwards. sum_kbn(A) should be implemented in terms of sum_kbn(identity, A), not the other way around. This way there will be only one implementation of the sum.

Reply:
Do you mean that I should follow this pattern?
```julia
function sum_kbn(f, A)
     ...
end
sum_kbn(a) = sum_kbn(identity, A)
```